embedded-graphics/src/primitives/rounded_rectangle.rs

```diff
@@ -0,0 +1,361 @@
+//! The rectangle primitive. Also good for drawing squares.
+
+use crate::{
+    drawable::{Drawable, Pixel},
+    geometry::{Dimensions, Point, Size},
+    pixelcolor::PixelColor,
+    primitives::{
+        ellipse_quadrant::{self, EllipseQuadrant, Quadrant, StyledEllipseQuadrantIterator},
+        rectangle::{self, Rectangle, StyledRectangleIterator},
+        ContainsPoint, Primitive,
+    },
+    style::{PrimitiveStyle, Styled},
+    transform::Transform,
+    DrawTarget,
+};
+
+/// Rounded rectangle primitive
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct RoundedRectangle {
+    /// The base rectangle
+    pub rectangle: Rectangle,
+
+    corners: [EllipseQuadrant; 4],
```

Comment:
I decided not to make this `pub` so we don't have to export `EllipseQuadrant`. If you think it's a good idea export it, I can make it pub, but it might be a bit of an odd thing to export as an e-g primitive?

Reply:
I haven't had time to look at the code in detail, but does this `EllipseQuadrant` primitive offer many advantages? Would splitting the code into `RoundedRectangle` and `EllipseQuadrant` make it harder to implement future optimizations, like drawing a rounded rectangle by using scanlines? (A scanline could start in one `EllipseQuadrant` and end in antoher)

> If you think it's a good idea export it, I can make it pub, but it might be a bit of an odd thing to export as an e-g primitive?

I agree that exporting `EllipseQuadrant` would be odd. If possible I would like to revert to the earlier version that used a `Size` to specify the corner radii. Not being able to access the `corners` field would make this primitive different than all others and we would need to add getters and setters to access and change the radii.

Regardless of whether we use `EllipseQuadrant` or `Size` can we also use a struct instead of an array for the radii?
```rust
pub struct CornerRadii {
    pub top_left: Size,
    pub top_right: Size,
    pub bottom_left: Size,
    pub bottom_right: Size,
}
```
IMO it's easier to use than having to remember which array index corresponds to which corner. And we could also add additional constructors like `CornerRadii::new_equal(size)`, `CornerRadii::new_hv(h_size, v_size)` or `CornerRadii::new_circular(10)`.

Reply:
I've changed to a struct which is much better, thanks for the suggestion.

> does this EllipseQuadrant primitive offer many advantages?

Mostly code separation. The `ellipse_quadrant.rs` file is pretty long and it has its own unit tests, etc. 

> Would splitting the code into RoundedRectangle and EllipseQuadrant make it harder to implement future optimizations, like drawing a rounded rectangle by using scanlines?

If we leave it as an internal implementation detail, we can optimise it out later. 

The other option is ripping out bits of `Ellipse` and reusing them directly in `rounded_rectangle::Points` and `rounded_rectangle::StyledRoundedRectangleIterator`. That way I think we could get rid of `EllipseQuadrant` completely. I might take a look at this approach.